src/types_.py

```diff
@@ -10,6 +10,35 @@
 from urllib.parse import urlparse
 
 
+@dataclasses.dataclass
+class MigrationInputs:
+    """Configurable parameters for migration mode.
+
+    Attrs:
+        github_access_token: A Personal Access Token(PAT) or access token with repository access.
+    """
+
+    github_access_token: str | None
+
+
+@dataclasses.dataclass
+class ReconcileInputs:
+    """Configurable parameters for reconcile mode.
+
+    Attrs:
+        dry_run: If enabled, only log the action that would be taken.
+        delete_pages: Whether to delete pages that are no longer needed.
+    """
+
+    dry_run: bool
+    delete_pages: bool
+
+
+@dataclasses.dataclass
+class UserInputs(ReconcileInputs, MigrationInputs):
```

Comment:
Is the class hierachy strictly necessary? What if an input is relevant for both in the future?

Reply:
My thought process for this was to keep a separation between flows, which would help track what inputs are needed for which flow.
I think it should have been a `BaseInputs` that contains inputs required for both flows (discourse params) and `ReconcileInputs` and `MigrationInputs` building on top of the `BaseInputs`. This would then flow well with the comment above regarding splitting `main` into different parts.

Reply:
I also wouldn't mind if they were all in the same class, class hierarchy adds complexity and should have significant benefits associated with it as a counter-balance to the increased complexity